cohttp-lwt/src/client.ml

```diff
@@ -53,21 +53,38 @@ module Make (IO : S.IO) (Net : S.Net with module IO = IO) (Time : Mirage_time.S)
       ; waiting :req_resr Queue.t
       ; condition :unit Lwt_condition.t (* watching queues *)
       ; finalise :(t -> unit Lwt.t)
+      ; id :int
       }
 
     let length connection =
       Queue.length connection.in_flight +
       Queue.length connection.waiting
 
+    let to_string connection =
+      Printf.sprintf "Connection %u (%u %s)"
+        connection.id
+        (length connection)
+        (match connection.state with
+        |Connecting _ -> "Connecting"
+        |Full _ -> "Full"
+        |Closing _ -> "Closing"
+        |Half _ -> "Half"
+        |Closed -> "Closed"
+        |Failed _ -> "Failed")
+
+    let log pipeline s = prerr_endline (to_string pipeline ^ ": " ^ s)
```

Comment:
cohttp-lwt uses logs for logging (it is already linking logs and logs.lwt), we should stick to that